File: frontend/src/engine/dice.ts
import type { DiceState } from '../store/gameStore'

export function rollDice(): DiceState {
  const d1 = Math.ceil(Math.random() * 6)
  const d2 = Math.ceil(Math.random() * 6)
  return {
    values: [d1, d2],
    remaining: d1 === d2 ? [d1, d1, d1, d1] : [d1, d2],
  }
}

File: backend/app/engine/dice.py
import random
from .state import DiceState


def roll_dice() -> DiceState:
    d1 = random.randint(1, 6)
    d2 = random.randint(1, 6)
    remaining = [d1, d1, d1, d1] if d1 == d2 else [d1, d2]
    return DiceState(values=[d1, d2], remaining=remaining)
